docs/content/developers/scopes.md

```diff
@@ -0,0 +1,256 @@
+# Scopes
+
+Een scope in deze context is een label, bijvoorbeeld `zds.scopes.zaken.lezen`
+waar een bepaalde betekenis aan toegekend is. We gebruiken scopes om
+toegangsniveaus uit te drukken.
+
+Toegangscontrole in de APIs speelt op verschillende niveaus, waarbij met
+gegevens binnen de relevante data rekening moet gehouden worden.
+
+In de meest eenvoudige vorm kan een scope bepalen of je al dan niet toegang
+hebt tot een operatie binnen een resource. Een `alleen-lezen` scope zou je
+bijvoorbeeld toegang kunnen geven op `zaak_read` en `zaak_list`, maar toegang
+verbieden op `zaak_create`, `zaak_update` en `zaak_partial_update`.
+
+Zodra je scopes gaat onderkennen die meer businesslogica bevatten, zoals scopes
+koppelen aan de rol van een betrokkene, dan kan het zijn dat binnen collecties
+een andere (sub)set van records teruggegeven wordt doordat de scope dit inperkt.
+Een concreet voorbeeld hiervan kunnen zaakobjecten zijn die wel of niet
+privacygevoel zijn. Panden zouden bijvoorbeeld altijd teruggegeven kunnen
+worden (voor elke scope), maar gerelateerde natuurlijke personen zouden
+weggelaten kunnen worden voor een 'publieke' scope.
+
+Tot slot onderkennen we nog een niveau waarop gegevens ingehouden kunnen worden.
+Indien je een record mag lezen, dan kan het zijn dat je niet alle attributen
+binnen de record mag lezen. Bijvoorbeeld een klantcontactcentrummedewerker
+zou de voornaam + naam van de zaakinitiator mogen lezen, maar niet het BSN
+van deze persoon.
+
+De uitdaging bestaat erin om een set van scopes te definieren die beheersbaar
+is, maar toch toelaat om om te gaan met:
+
+- vertrouwelijkheidsaanduiding
+- roltypen
+- AVG/privacy
+
+## Algemeen
+
+_Inzichten van Arjan_:
+
+De mogelijkheden voor een gebruiker om gegevens van een object - zoals
+'Zaak 1234' of 'Document Abcd' - te mogen raadplegen en/of bewerken hangen
+m.i. af van (minimaal) vier factoren:
+
+1. **_De categorie van het object_**.
+   Bijvoorbeeld het zaaktype van een zaak, of een zaak al dan niet beeindigd
+   is, etc. Zo zal een gebruiker veelal slechts zaken van bepaalde zaaktypen
+   mogen raadplegen, of alleen onderhanden zaken.
+
```

Comment:
Kunnen scopes dynamisch zijn, d.w.z. kunnen deze worden toegevoegd als er een nieuwe rol of zaaktype ontstaat? Achtergrond van mijn vraag is dat autorisatie in een zaaksysteem op basis van zaaktypen kan worden gedaan. Dan moet je denken aan rechten om zaken van bepaald zaaktype te mogen inzien, creëren, bewerken.

Reply:
Dat is 1 van de grote vragen om te beantwoorden. In principe zijn scopes zelf (de labels) statisch, dus om dat van zaaktype te kunnen laten afhangen moet er een mechanisme zijn.

Reply:
@Hugo-ter-Doest scopes zelf zijn statisch, maar we voegen naast de scopes claim ook een `zaaktypes` claim toe die dynamisch is. Op dezelfde manier wil ik met `vertrouwelijkheidaanduiding` claim gaan werken.